posthog/api/action.py

```diff
@@ -369,67 +341,51 @@ def _serialize_people(self, id: str, name: str, people: QuerySet, request: reque
             'count': len(people_dict)
         }
 
-    def _process_entity_for_events(self, entity, entity_type=None, order_by="-id") -> QuerySet:
-        if entity_type == TREND_FILTER_TYPE_ACTIONS:
-            return Event.objects.filter_by_action(action=entity, order_by=order_by)
-        elif entity_type == TREND_FILTER_TYPE_EVENTS:
-            return Event.objects.filter_by_event_with_people(event=entity['id'], team_id=self.request.user.team_set.get().id, order_by=order_by)
+    def _process_entity_for_events(self, entity: Entity, team: Team, order_by="-id") -> QuerySet:
+        if entity.type == TREND_FILTER_TYPE_ACTIONS:
+            events = Event.objects.filter(action__pk=entity.id).add_person_id(team.pk)
```

Comment:
I'm assuming you thought about this and went this route rather than trying to cast the entity to an action, but my initial thought is that we could add a property or function on the Entity model that provides a casting to action if possible so that we can reuse `filter_by_action` here rather than largely repeating the logic.

Reply:
Hm agree it's a little annoying to repeat the logic, but couldn't find a way of doing this without triggering another Action query. I also don't think having the Entity object carry around the loaded Action object makes sense. The solution was going to be more complicated than repeating these 3 lines of code which is why I went for this.

However I don't feel strongly either way

Reply:
Oh I see. It's nit so I think it's okay